api/src/play_asset_delivery_request_tracker.gd

```diff
@@ -62,6 +62,8 @@ func lookup_request(signal_id : int) -> PlayAssetDeliveryRequest:
 func unregister_request(signal_id : int) -> void:
 	_request_tracker_mutex.lock()
 	if signal_id in _signal_id_to_request_map:
-		_signal_id_to_request_map.erase(signal_id)
+		var erase_success = _signal_id_to_request_map.erase(signal_id)
+		if not erase_success:
+			push_error("Erase " + str(signal_id) + " key from _signal_id_to_request_map failed!")
```

Comment:
This would work, but only if the request tracker considers this case an error. Is there anywhere in the code where you might unregister a request without knowing if it's registered?

Reply:
By design we should never unregister a request without knowing if it is registered. Currently there is a bug in ```_on_fetch_success``` that can cause us to call erase [twice](https://github.com/Arivval/Godot-Play-Asset-Delivery/blob/1a0ec291285cd1ba02f6ba8206b5d29fa2e2a30c/api/src/model/request/play_asset_pack_fetch_request.gd#L98), and it has been fixed in PR #29.

Reply:
I think it is better for us to consider this as a warning rather than error, since if we call ```erase()``` on an already erased key, we still achieve this desired state, which does not necessarily count as an error; Having this as a warning can help us to know what is actually happening when we call ```erase()```.
I just modified the ```push_error()``` to ```push_warning()```.

Reply:
Discussed with Jake about this offline: Since it is impossible to call ```erase()``` on non-existing key entries, and developer should not worry about this case, we are removing the error handling code for ```erase()```.